### PR TITLE
Add get/setmempoolfree rpc commands

### PR DESCRIPTION
**Pull request by alexlyp**
getmempoolfee RPC command returns currently set mempool fee variables. (RelayFee, MinFee and SkipFeeLocal bool)

setmempoolfee <relayfee> <minfee> <skipfeelocal> RPC allows the user to set their daemons mempool fee rules.

### CODE REVIEW

**Comment by jcvernaleo** [comment]
jcv@triforce dcrd $ go test
--- FAIL: TestHelp (0.00s)
        rpcserverhelp_test.go:18: RPC handler defined for method 'getmempoolfee' without also specifying result types
        rpcserverhelp_test.go:18: RPC handler defined for method 'setmempoolfee' without also specifying result types
        rpcserverhelp_test.go:45: Failed to generate help for method 'getmempoolfee': no result types specified for method getmempoolfee
        rpcserverhelp_test.go:45: Failed to generate help for method 'setmempoolfee': no result types specified for method setmempoolfee
FAIL
exit status 1
FAIL    github.com/decred/dcrd  0.029s

File: dcrjson/dcrdextcmds.go [hunk 1]
```diff
@@ -84,4 +106,6 @@ func init() {
 	MustRegisterCmd("rebroadcastmissed", (*RebroadcastMissedCmd)(nil), flags)
 	MustRegisterCmd("rebroadcastwinners", (*RebroadcastWinnersCmd)(nil), flags)
 	MustRegisterCmd("ticketsforaddress", (*TicketsForAddressCmd)(nil), flags)
+	MustRegisterCmd("getmempoolfee", (*GetMempoolFeeCmd)(nil), flags)
+	MustRegisterCmd("setmempoolfee", (*SetMempoolFeeCmd)(nil), flags)
```
Comment (jcvernaleo) [inline dcrjson/dcrdextcmds.go]: Probably want these in the right place alphabetically.

File: rpcserver.go [hunk 2]
```diff
@@ -4120,6 +4137,40 @@ func handleSetGenerate(s *rpcServer, cmd interface{}, closeChan <-chan struct{})
 	return nil, nil
 }
 
+// handleGetMempoolFee implements the getmempoolfee command. We will return
```
Comment (jcvernaleo) [inline rpcserver.go]: I think this comment is just copied from the func above, should update for this function.

**Comment by cjepson** [comment]
SkipFeeLocal should cause all transactions submitted through the RPC via sendrawtransaction to skip all fees checks. You will need to add a boolean flag to blockManager.ProcessTransaction which is passed down the outgoing functions until it reaches the mempool transactions. The flag can be called something like "skipFeesCheck", similar to the other flags passed to ProcessTransaction.

The skipFeesCheck flag in handleSendRawTransaction should only be true if the flag in the mempool is set to enable it. 

For example, in handleSendRawTransaction, you'd add:
`
	skipFeeLocal := mempool.SkipFeeLocal()
	err = s.server.blockManager.ProcessTransaction(tx, false, false, skipFeeLocal)
`

Everywhere else in the codebase this flag would be set to false and fees checks should always be done. If the flag passed to maybeAcceptTransaction in the mempool is set to true, it should skip all these checks: https://github.com/decred/dcrd/blob/master/mempool.go#L1577-L1664

Consider the use of an if or labeled GOTO statement to skip these checks in mempool for the flag.

Mining also needs to know that these transactions are high priority despite lack of fees. You will need to maintain a list of these transactions (probably a map[chainhash.Hash]struct{}) so that when they are iterated through on line 1276-1290 of mining.go, they are considered to have higher fees than they do. This is not a very good fix and is used to inject stake transactions into a block. Soon we should refactor it, but we can use the manipulation of fee per KB there to artificially push these transactions to the front of the queue. When going through this portion of the loop, it should check the map to see if the transaction is there. If it is, it should tweak the feesPerKb to it's artificially high.

You will also have to skip the check in mining.go on L1445-1458, so that the transaction is included even if the fee is too low.

This should be tested to ensure it works after on simnet. You can make the minFees very high for mempool, toggle the flag for sendrawtransaction, submit a low or no fee tx, then mine and see if the transaction is included in a block.

**Comment by jcvernaleo** [comment]
goclean.sh catches a comment issue:
dcrjson/dcrdextcmds.go:101:1: comment on exported function NewrSetMempoolFeeCmd should be of the form '"NewrSetMempoolFeeCmd' '..."'

**Comment by alexlyp** [comment]
This PR has been overwritten by other various upgrades recently. Closing

**Pull request closed without merging**